flink-end-to-end-tests/test-scripts/test_kubernetes_embedded_job.sh

```diff
@@ -49,7 +54,4 @@ kubectl wait --for=condition=complete job/flink-job-cluster --timeout=1h
 kubectl cp `kubectl get pods | awk '/task-manager/ {print $1}'`:/cache/${OUTPUT_FILE} ${OUTPUT_VOLUME}/${OUTPUT_FILE}
 
 check_result_hash "WordCount" ${OUTPUT_VOLUME}/${OUTPUT_FILE} "${RESULT_HASH}"
-
-if [ $? != 0 ];then
-  debug_copy_and_show_logs
-fi
+SUCCEEDED=$?
```

Comment:
I recently got to know bash traps, which are basically signal handlers.
You can do `trap debug_and_show_logs EXIT`, which will run `debug_and_show_logs` whenever the script exists.

Reply:
We have already use the `on_exit cleanup` in `common_kubernetes.sh`. So i put the `debug_and_show_logs` in `cleanup`, it will always be called before the script exits.

Reply:
True. You could register `on_exit debug_and_show_logs` before sourcing `common_kubernetes.sh`.

But I'm fine with your solution. I just wanted to mention it so that you know it exists :)